### PR TITLE
Add host_port parameter to marathon.yaml

### PR DESCRIPTION
**Pull request by EvanKrall**

### CODE REVIEW

**Comment by Rob-Johnson** [comment]
I think you'll need to update the json schema in https://github.com/Yelp/paasta/blob/master/paasta_tools/cli/schemas/marathon_schema.json

File: paasta_tools/marathon_tools.py [hunk 1]
```diff
@@ -377,10 +377,19 @@ def format_marathon_app_dict(self):
...
             ]
+        else:
+            complete_config['port_definitions'] = [
+                {
+                    'port': self.get_host_port(),
+                    'protocol': 'tcp',
+                },
+            ]
+            # Without this, we may end up with multiple containers requiring the same port on the same box.
+            complete_config['require_ports'] = True
```
Comment (solarkennedy) [inline paasta_tools/marathon_tools.py]: whoa, I didn't think you need this

**Comment by EvanKrall** [comment]
@Rob-Johnson Good catch. I always forget about the schema when I add new keys.

File: paasta_tools/marathon_tools.py (hunk 1 above)
Comment (EvanKrall) [inline paasta_tools/marathon_tools.py]: Actually, that's an interesting point. We might be able to do without it, so long as we add hostname:UNIQUE constraints.

requirePorts does restrict us to only using ports that are in the mesos offers (e.g. 31000-31999).

Should I drop it?

Comment (EvanKrall) [inline paasta_tools/marathon_tools.py]: Hrm. Actually, if I drop this line, the tests don't pass. Our `marathon_services_running_here` is looking at the tasks listed in the mesos slave, and the ports attached to them, so unfortunately if marathon doesn't grab a port from the offer, we won't discover the tasks correctly.

Without `requirePorts`, Marathon still grabs an arbitrary port from the offer, which is what will show up in the mesos slave info.


Docker env with `"requirePorts": false`:

```
            "Env": [
                "PAASTA_DOCKER_IMAGE=busybox",
                "MARATHON_APP_VERSION=2017-03-23T17:35:52.291Z",
                "MARATHON_APP_RESOURCE_CPUS=0.1",
                "MARATHON_PORTS=31534",
                "MARATHON_APP_RESOURCE_GPUS=0",
                "MARATHON_PORT=31534",
                "MARATHON_APP_DOCKER_IMAGE=busybox",
                "PAASTA_SERVICE=bounce",
                "MARATHON_PORT_31337=31534",
                "MARATHON_PORT0=31534",
                "MESOS_TASK_ID=bounce.test1.newapp.confighash.29590637-0fef-11e7-a470-0242ac110007",
                "PAASTA_CLUSTER=testcluster",
                "MARATHON_APP_RESOURCE_MEM=100.0",
                "MARATHON_APP_RESOURCE_DISK=10.0",
                "MARATHON_APP_LABELS=",
                "PAASTA_INSTANCE=test1",
                "MARATHON_APP_ID=/bounce.test1.newapp.confighash",
                "MARATHON_HOST=750a7e66cd63",
                "MESOS_SANDBOX=/mnt/mesos/sandbox",
                "MESOS_CONTAINER_NAME=mesos-2608f33a-bcd8-4413-a277-26398f23e743-S1.ca0c5734-e5f1-45b7-8133-bcafe82f7f55",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
```

Docker env with `"requirePorts": true`:

```
            "Env": [
                "PAASTA_DOCKER_IMAGE=busybox",
                "MARATHON_APP_VERSION=2017-03-23T17:40:55.182Z",
                "MARATHON_APP_RESOURCE_CPUS=0.1",
                "MARATHON_PORTS=31337",
                "MARATHON_APP_RESOURCE_GPUS=0",
                "MARATHON_PORT=31337",
                "MARATHON_APP_DOCKER_IMAGE=busybox",
                "PAASTA_SERVICE=bounce",
                "MARATHON_PORT_31337=31337",
                "MARATHON_PORT0=31337",
                "MESOS_TASK_ID=bounce.test1.newapp.confighash.ddde4029-0fef-11e7-a470-0242ac110007",
                "PAASTA_CLUSTER=testcluster",
                "MARATHON_APP_RESOURCE_MEM=100.0",
                "MARATHON_APP_RESOURCE_DISK=10.0",
                "MARATHON_APP_LABELS=",
                "PAASTA_INSTANCE=test1",
                "MARATHON_APP_ID=/bounce.test1.newapp.confighash",
                "MARATHON_HOST=73fe825ebd54",
                "MESOS_SANDBOX=/mnt/mesos/sandbox",
                "MESOS_CONTAINER_NAME=mesos-2608f33a-bcd8-4413-a277-26398f23e743-S0.a3595e96-0237-4ffb-a51a-6ac4b903755c",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
```

Comment (solarkennedy) [inline paasta_tools/marathon_tools.py]: When I said that, I meant "wow, it isn't implied that the ports are required if you ask them?"

ship on

**Pull request merged**